packages/protocol/test/governance/validators.ts

```diff
@@ -1016,53 +1068,116 @@ contract('Validators', (accounts: string[]) => {
+            // @ts-ignore Broken typechain typing for bytes
+            await assertRevert(validators.updateEcdsaKey(validator, signer, newPublicKey))
+          })
+        })
+      })
+
+      describe('when not called by the registered `Accounts` contract', () => {
+        describe('when the public key matches the signer', () => {
+          let newPublicKey: string
+          const signer = accounts[9]
+          it('should revert', async () => {
+            newPublicKey = await addressToPublicKey(signer, web3)
+            // @ts-ignore Broken typechain typing for bytes
+            await assertRevert(validators.updateEcdsaKey(validator, signer, newPublicKey))
+          })
+        })
+      })
+    })
+  })
+
+  describe('#updateBlsKey()', () => {
+    const newBlsPublicKey = web3.utils.randomHex(48)
+    const newBlsPoP = web3.utils.randomHex(96)
     describe('when called by a registered validator', () => {
       const validator = accounts[0]
       beforeEach(async () => {
         await registerValidator(validator)
       })
 
-      describe('when the public keys data is the right length', () => {
+      describe('when the keys are the right length', () => {
         let resp: any
         beforeEach(async () => {
           // @ts-ignore Broken typechain typing for bytes
-          resp = await validators.updatePublicKeysData(newPublicKeysData)
+          resp = await validators.updateBlsKey(newBlsPublicKey, newBlsPoP)
         })
 
-        it('should set the validator public keys data', async () => {
+        it('should set the validator bls public key', async () => {
           const parsedValidator = parseValidatorParams(await validators.getValidator(validator))
-          assert.equal(parsedValidator.publicKeysData, newPublicKeysData)
+          assert.equal(parsedValidator.blsKey, newBlsPublicKey)
         })
 
-        it('should emit the ValidatorPublicKeysDataUpdated event', async () => {
+        it('should emit the ValidatorBlsKeyUpdated event', async () => {
           assert.equal(resp.logs.length, 1)
           const log = resp.logs[0]
           assertContainSubset(log, {
-            event: 'ValidatorPublicKeysDataUpdated',
+            event: 'ValidatorBlsKeyUpdated',
             args: {
               validator,
-              publicKeysData: newPublicKeysData,
+              blsKey: newBlsPublicKey,
             },
           })
         })
       })
 
-      describe('when the public keys data is too long', () => {
+      describe('when the public key is not 48 bytes', () => {
         it('should revert', async () => {
           // @ts-ignore Broken typechain typing for bytes
-          await assertRevert(validators.updatePublicKeysData(newPublicKeysData + '00'))
+          await assertRevert(validators.updateBlsKey(newBlsPublicKey + '01', newBlsPoP))
         })
       })
 
-      describe('when the public keys data is too short', () => {
```

Comment:
It would be helpful to also have a check to make sure we revert if the PoP is invalid (e.g using `newBlsPoP.slice(0, -2) + '01'`)

Reply:
Since we actually don't have this precompile in ganache we can't check this yet 🙁